packages/pages/src/common/src/function/internal/validateServerlessFunctionModuleInternal.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  ServerlessFunctionConfigInternal,
+  ServerlessFunctionModuleInternal,
+} from "./types.js";
+import {
+  ServerlessFunctionModule,
+  ServerlessFunctionConfig,
+} from "../types.js";
+
+export const validateServerlessFunctionDefaultExport = (
+  filepath: string,
+  serverlessFunction: ServerlessFunctionModule<any>
+) => {
+  if (!serverlessFunction.default) {
+    throw new Error(`${filepath} is missing an default export function.`);
+  }
+};
+
+export const validateServerlessFunctionModuleInternal = (
```

Comment:
We don't need to validate the internal domain object, only the external one. If the external passes validation then the internal one we construct should be fine.